### PR TITLE
[Ploopy Nano] Add scrolling on numlock and bootloader mode on caps lock. Mouse acceleration optional in keymap. 

### PR DESCRIPTION
**Pull request by squk**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by drashna** [comment]
You have some merge conflicts that need to be addressed here.

**Comment by squk** [comment]
> You have some merge conflicts that need to be addressed here.

Resolved.

**Comment by squk** [comment]
Bump :)

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/ploopyco/trackball_nano/trackball_nano.c [hunk 1]
```diff
@@ -57,14 +64,123 @@ uint16_t          dpi_array[] = PLOOPY_DPI_OPTIONS;
...
+        // But the x*x cancels the sign, so we need to negate it if the input value is negative.
+        x = (x > 0 ? x*x/16+x : -x*x/16+x);
+        y = (y > 0 ? y*y/16+y : -y*y/16+y);
+    }
+
+    mouse_report->x = x;
+    mouse_report->y = y;
+}
+
+__attribute__((weak)) void process_mouse(report_mouse_t* mouse_report) {
```
[the remark this answers is not in the document]
Comment (squk) [reply keyboards/ploopyco/trackball_nano/trackball_nano.c:127]: GitHub is only showing 4 lines selected. Could you give me the line numbers :)

**Pull request closed without merging**